ankihub/gui/deckbrowser.py

```diff
@@ -99,39 +101,26 @@ def on_button_clicked(button_index: int) -> None:
 def _setup_update_subdeck_due_date(menu: QMenu, subdeck_did: DeckId) -> None:
     action = menu.addAction("Ankihub: Update due date")
 
-    subdeck_config = config.get_block_exam_subdeck_config(subdeck_did)
-
-    action.setEnabled(subdeck_config is not None)
+    initial_due_date = config.get_block_exam_subdeck_due_date(subdeck_did)
 
-    if subdeck_config:
-        action.setToolTip("Change the due date of this subdeck.")
-        qconnect(action.triggered, lambda: _open_date_picker_dialog_for_subdeck(subdeck_config))
-    else:
-        action.setToolTip("This option is only available for subdecks created with SmartSearch")
+    action.setToolTip("Change the due date of this subdeck.")
+    qconnect(
+        action.triggered,
+        lambda: _open_date_picker_dialog_for_subdeck(subdeck_did, initial_due_date),
+    )
 
 
 def _setup_remove_block_exam_subdeck(menu: QMenu, subdeck_did: DeckId) -> None:
     action = menu.addAction("Ankihub: Remove subdeck")
 
-    subdeck_config = config.get_block_exam_subdeck_config(subdeck_did)
-
-    action.setEnabled(subdeck_config is not None)
-
-    if subdeck_config:
-        action.setToolTip("Deletes the subdeck and moves all notes back into the main deck.")
-        qconnect(action.triggered, lambda: _open_remove_block_exam_subdeck_dialog(subdeck_config))
-    else:
-        action.setToolTip("This option is only available for subdecks created with SmartSearch")
+    action.setToolTip("Deletes the subdeck and moves all notes back into the main deck.")
+    qconnect(action.triggered, lambda: _open_remove_block_exam_subdeck_dialog(subdeck_did))
 
 
 def _initialize_subdeck_context_menu_actions(menu: QMenu, deck_id: int) -> None:
-    # Only show the menu actions for descendants of AnkiHub decks
-    is_descendant_of_ah_deck = (
-        get_ah_did_of_deck_or_ancestor_deck(DeckId(deck_id)) is not None
-        and aqt.mw.col.decks.parents(DeckId(deck_id))  # Ensure it's not a top-level deck
-    )
-    if not is_descendant_of_ah_deck:
-        return
+    # Only show the menu actions for subdecks (not top-level decks)
+    if not aqt.mw.col.decks.parents(DeckId(deck_id)):
+        return  # Skip top-level decks
 
     def on_access_granted(_: dict) -> None:
         menu.setToolTipsVisible(True)
```

Comment:
:seedling: It'd be cool to check the permission on hover to avoid delay on showing options


<video src="https://github.com/user-attachments/assets/04fb247a-434b-4233-850f-a3a11b69a954" width="200" />

Reply:
> It'd be cool to check the permission on hover to avoid delay on showing options

I'm not sure we can do that, because the options would then initially appear enabled?

 We have an improvement task for dealing with the delay: https://ankihub.atlassian.net/jira/software/c/projects/NRT/boards/41/backlog?selectedIssue=NRT-424